### PR TITLE
Analytics rationalisation and update

### PR DESCRIPTION
**Pull request by injms**
### Context
The analytics audit showed that there was a lot of tracking where it wasn't needed, and an absence of tracking where it was needed. 

### Changes proposed in this pull request
 * Events no longer fire when links the header and the footer of the page are clicked
 * Events no longer fire when a register is sorted
 * The feedback form now includes the 'No' sub-options, and only fires when the form is submitted
 * Accordions only fire an event when opened, not when closed
 * Unsuccessful searches now fire an event to let us know what query wasn't found

### Guidance to review
(Apologies for the mass of linting changes that have become inextricably linked to this PR).
This has been tested in all of the supported browsers - but extra eyes are always welcome!

### CODE REVIEW

File: app/assets/javascripts/accordion-with-descriptions.js [hunk 1]
```diff
@@ -18,7 +17,7 @@
       var $openOrCloseAllButton;
       var GOVUKServiceManualTopic = serviceManualTopicPrefix();
 
-      //addOpenCloseAllButton();
```
Comment (gidsg) [inline app/assets/javascripts/accordion-with-descriptions.js]: If this is not being used can we delete it rather than commenting it out?

**Comment by gidsg** [comment]
Can we use XO linter as this is the GDS way: https://gds-way.cloudapps.digital/manuals/programming-languages/nodejs/#source-formatting-and-linting

File: app/assets/javascripts/accordion-with-descriptions.js [hunk 2]
```diff
@@ -210,14 +208,14 @@
         });
       }
 
-      function openStoredSections($section) {
+      function openStoredSections ($section) {
         toggleSection($section);
         toggleIcon($section);
         toggleState($section.parent().find('.subsection__button'));
         // setOpenCloseAllText();
```
Comment (gidsg) [inline app/assets/javascripts/accordion-with-descriptions.js]: As per above can we uncomment or delete?

File: app/assets/javascripts/accordion-with-descriptions.js [hunk 3]
```diff
@@ -210,14 +208,14 @@
...
       }
 
-      function setOpenCloseAllText() {
+      function setOpenCloseAllText () {
         var openSubsections = $('.subsection--is-open').length;
         // Find out if the number of is-opens == total number of sections
```
Comment (gidsg) [inline app/assets/javascripts/accordion-with-descriptions.js]: I know this PR has not changed this, but if this is still an open question can we prefix with `TODO: ` so it is better recognised by IDEs?

File: app/views/layouts/_analytics.html.erb [hunk 4]
```diff
@@ -1,17 +1,20 @@
 <!-- Google Analytics -->
 <script>
+/* global ga, window */
```
Comment (gidsg) [inline app/views/layouts/_analytics.html.erb]: since this is refactoring the way we instanciate the GA JS, could we pull this out into a separate PR to split it out from the event tracking changes?

File: app/assets/javascripts/accordion-with-descriptions.js (hunk 1 above)
Comment (injms) [reply app/assets/javascripts/accordion-with-descriptions.js]: Sure thing - I'll bundle this into another 'neatening code' type pull request separate to this one.

File: app/assets/javascripts/accordion-with-descriptions.js (hunk 2 above)
Comment (injms) [reply app/assets/javascripts/accordion-with-descriptions.js]: Okay - I'll put that in a separate pull request.

File: app/assets/javascripts/accordion-with-descriptions.js (hunk 3 above)
Comment (injms) [reply app/assets/javascripts/accordion-with-descriptions.js]: Okay - this linter has highlighted a lot of stuff that isn't mine...

File: app/views/layouts/_analytics.html.erb (hunk 4 above)
Comment (injms) [reply app/views/layouts/_analytics.html.erb]: Good point - I'll cherry pick and put in a separate pull request.

**Comment by injms** [comment]
This has snowballed, as the accidental reformatting has highlighted things that neatened up. I'm closing this, and I'll put in a set of pull requests that will fix all of @gidsg comments and suggestions.
